tiatoolbox/models/engine/engine_abc.py

```diff
@@ -1,21 +1,586 @@
+                image tiles.
+            labels (list | None):
+                List of labels. If using `tile` or `wsi` mode, then only
+                a single label per image tile or whole-slide image is
+                supported.
+            patch_mode (bool):
+                Whether to treat input image as a patch or WSI.
+                default = True.
+            ioconfig (IOPatchPredictorConfig):
+                IO configuration.
+            save_dir (str or pathlib.Path):
+                Output directory when processing multiple tiles and
+                whole-slide images. By default, it is folder `output`
+                where the running script is invoked.
+            overwrite (bool):
+                Whether to overwrite the results. Default = False.
+            output_type (str):
+                The format of the output type. "output_type" can be
+                "dict", "array", "AnnotationStore", "DataFrame" or "json".
+                Default is "AnnotationStore".
+            **kwargs (dict):
+                Keyword Args to update :class:`EngineABC` attributes.
+
+        Returns:
+            (:class:`numpy.ndarray`, dict):
+                Model predictions of the input dataset. If multiple
+                image tiles or whole-slide images are provided as input,
+                or save_output is True, then results are saved to
+                `save_dir` and a dictionary indicating save location for
+                each input is returned.
+
+                The dict has the following format:
+
+                - img_path: path of the input image.
+                - raw: path to save location for raw prediction,
+                  saved in .json.
+                - merged: path to .npy contain merged
+                  predictions if `merge_predictions` is `True`.
+
+        Examples:
+            >>> wsis = ['wsi1.svs', 'wsi2.svs']
+            >>> predictor = EngineABC(
+            ...                 pretrained_model="resnet18-kather100k")
+            >>> output = predictor.run(wsis, patch_mode=False)
+            >>> output.keys()
+            ... ['wsi1.svs', 'wsi2.svs']
+            >>> output['wsi1.svs']
+            ... {'raw': '0.raw.json', 'merged': '0.merged.npy'}
+            >>> output['wsi2.svs']
+            ... {'raw': '1.raw.json', 'merged': '1.merged.npy'}
+
+        """
+        for key in kwargs:
+            setattr(self, key, kwargs[key])
+
+        self._validate_input_numbers(images=images, masks=masks, labels=labels)
+        self.images = self._validate_images_masks(images=images)
+
+        if masks is not None:
+            self.masks = self._validate_images_masks(images=masks)
+
+        self.labels = labels
+
+        # if necessary Move model parameters to "cpu" or "gpu" and update ioconfig
+        self._ioconfig = self._load_ioconfig(ioconfig=ioconfig)
+        self.model = model_to(model=self.model, on_gpu=self.on_gpu)
```

Comment:
```suggestion
        self.model = model_to(model=self.model, device=self.device)
```

Reply:
Addressed in this commit https://github.com/TissueImageAnalytics/tiatoolbox/pull/635/commits/d970f49baeb08badedd2d1d214354348b1ffd6cd